From: SJ Park <sj@kernel.org>
Date: Tue, 23 Jun 2026 07:23:49 -0700
Subject: [PATCH] ==== prep for access monitoring ====

mm/damon: introduce data access-as-a-data attribute

TL;DR: extend DAMON's data attributes monitoring system to support page
table accessed bit and PG_idle based access monitoring.

DAMON was initially introduced as data access monitor.  Users found data
access pattern becomes more useful when it is combined with other data
attributes such as belonging cgroups and backing page types.  For such
cases, DAMON has extended to support such data attributes monitoring in
addition to the original data access monitoring.  DAMON probes system
has introduced for the purpose.  Users set probes for filtering data
attributes of their interest.  For use cases where the primary interests
are the attributes but the access pattern, probe weights system has
introduced.  When it is used, DAMON applies its adaptive regions
adjustment based on the monitored data attributes.

However, DAMON stops access monitoring when the probe weights are used.
DAMON cannot optimally help users who have interests in both data
attributes and access pattern.  Data access can also be thought of as
another data attribute, though.  Extend the probe system to support data
access as a data attribute.

Introduce a new probe filter type, pgidle_unset.  It shows if the page
is not set as idle.  Specifically, it shows the page table accessed bit
and the PG_idle flag.  It can inform if the region is ever accessed.
But it cannot say when it is accessed.  To answer the second question,
introduce a new probe feature, prep actions.  Using the features, Users
can specify what preparation actions should be made to each region for
each probe.  DAMON executes the preparation actions for each sampling
interval, like it is doing the preparation for access check in the
access monitoring mode.  To help 'pgidle_unset' probe action use case,
'set_pgidle' preparation action is introduced together.  The action does
exactly what the access monitoring was doing: clearing the page table
accessed bits and setting the PG_idle flags.

Signed-off-by: SJ Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_pvnys32_ | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_pvnys32_

diff --git a/hkml_cv_bogus/hkml_cv_bogus_pvnys32_ b/hkml_cv_bogus/hkml_cv_bogus_pvnys32_
new file mode 100644
index 0000000000000..e69de29bb2d1d
-- 
2.47.3

